Automattic/wp-calypso
# Jetpack Connect: Move main header to a separate component

Some of the main components in Jetpack Connect are responsible for a lot of things happening, which results in them being large in terms of code. Further splitting them to subcomponents could help with maintaining them and introducing changes and new features.

This PR moves the Jetpack Connect main component header to a separate component. It consists of the logic for determining the main title and subtitle, containing the relevant copy, and rendering a `<FormattedHeader />` with them. This helps reduce the size of the `JetpackConnectMain` component by almost 15%.

To test:
* Checkout this branch.
* Compare the header title/subtitle in each of the following URLs to the production ones, and after each of them, delete the `jetpack_connect_selected_plan` cookie. There should be no difference with the production headers:
  * http://calypso.localhost:3000/jetpack/connect/
  * http://calypso.localhost:3000/jetpack/connect/pro
  * http://calypso.localhost:3000/jetpack/connect/premium
  * http://calypso.localhost:3000/jetpack/connect/personal
  * http://calypso.localhost:3000/jetpack/connect/install?flags=-jetpack/connect/remote-install (flag disabling is necessary because we're currently running Jetpack Remote Install on this same route)
